src/Microsoft.AspNetCore.Diagnostics.HealthChecks/HealthCheckMiddleware.cs

```diff
@@ -70,6 +71,13 @@ public async Task InvokeAsync(HttpContext httpContext)
 
             httpContext.Response.StatusCode = statusCode;
 
+            if (!_healthCheckOptions.SuppressCacheHeaders)
+            {
+                var headers = httpContext.Response.Headers;
+                headers[HeaderNames.CacheControl] = "no-store, no-cache";
+                headers[HeaderNames.Pragma] = "no-cache";
```

Comment:
[Elsewhere](https://github.com/aspnet/Security/blob/7b6c9cf0eeb149f2142dedd55a17430e7831ea99/src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs#L377-L379) we've also done Expires for thoroughness.